src/service.rs

```diff
@@ -235,19 +262,34 @@ where
     }
 }
 
-fn create_cookie<'a>(config: AxumSessionConfig, value: String) -> Cookie<'a> {
-    let mut cookie_builder = Cookie::build(config.cookie_name, value)
-        .path(config.cookie_path)
-        .secure(config.cookie_secure)
-        .http_only(config.cookie_http_only);
+fn create_cookie<'a>(
+    config: AxumSessionConfig,
+    value: String,
+    cookie_type: CookieType,
+) -> Cookie<'a> {
+    let mut cookie_builder = Cookie::build(
+        match cookie_type {
+            CookieType::Data => config.cookie_name,
+            CookieType::Accepted => config.accepted_cookie_name,
+        },
```

Comment:
I tend to use a utility method either on the enum or on the config to "get data based on enum case". Makes the code easier to follow!

```rust
impl AxumSessionConfig {
    fn cookie_name(&self, type: CookieType) -> &str {
        match type {
            CookieType::Data => &self.cookie_name,
            CookieType::Accepted => &self.accepted_cookie_name,
        }
    }
}

// or

impl CookieType {
    fn get_name(&self, config: &AxumSessionConfig) -> &str {
        match self {
            CookieType::Data => &config.cookie_name,
            CookieType::Accepted => &config.accepted_cookie_name,
        },
    }
}
```

Reply:
Same for max_age below!